.travis.yml

```diff
@@ -1,18 +1,34 @@
-sudo: false
+dist: trusty
+sudo: required
 language: java
 cache:
   directories:
   - $HOME/.m2
+addons:
+  apt:
+    packages:
+      - openjdk-6-jdk
 jdk:
   - openjdk6
   - openjdk7
-  - oraclejdk7
   - oraclejdk8
+  - oraclejdk9
 env:
   - FEDORA_VERSION=3.6.2
   - FEDORA_VERSION=3.7.0
   - FEDORA_VERSION=3.7.1
   - FEDORA_VERSION=3.8.1
+matrix:
+  allow_failures:
+    - jdk: oraclejdk9
+install:
+  - echo "Downloading Maven 3.0";
+  - wget https://archive.apache.org/dist/maven/binaries/apache-maven-3.0-bin.zip || travis_terminate 1
```

Comment:
Why install maven by hand here? Java project should have maven be default right? 
https://docs.travis-ci.com/user/languages/java/#Projects-Using-Maven

Reply:
Won't work. the shipped version does not work on our whole matrix and fails. See that references travis issues. You can verify locally if needed

Reply:
Again, this is installing an older version of maven for every build. I'd rather see us use newer versions when we can. It does not fail on our whole matrix, it only fails for openjdk6 for me.